src/Document.php

```diff
@@ -158,29 +126,27 @@ public function canCreateItem()
             }
         }
 
-        if (Document::canCreate()) {
+        if (self::canCreate()) {
             return parent::canCreateItem();
         }
         return false;
     }
 
-
     public function cleanDBonPurge()
     {
-
         $this->deleteChildrenAndRelationsFromDb(
             [
                 Document_Item::class,
             ]
         );
 
-       // UNLINK DU FICHIER
+       // Unlink/delete the file
         if (!empty($this->fields["filepath"])) {
             if (
                 is_file(GLPI_DOC_DIR . "/" . $this->fields["filepath"])
                 && !is_dir(GLPI_DOC_DIR . "/" . $this->fields["filepath"])
                 && (countElementsInTable(
-                    $this->getTable(),
+                    self::getTable(),
```

Comment:
I don't know if anything currently override Document class, but replacing `$this` should probably be replaced with `static::` rather  than `self::` the scope is not the same